modules/tinymce/src/plugins/template/test/ts/browser/TemplateSanityTest.ts

```diff
@@ -76,64 +77,80 @@ describe('browser.tinymce.plugins.template.TemplateSanityTest', () => {
+        templates: [{ title: 'a', description: 'b', content: `<html>${unparsedHtml}</html>` }],
+      });
+      await pPreviewAndAssertNoUnparsedContent(editor);
+    });
   });
 
-  const assertFnDoesNotReadUnsanitizedHtmlInDom = async (editor: Editor, fn: (unsanitizedHtml: string) => void | Promise<void>): Promise<void> => {
-    const fnReadsUnsanitizedHtmlInDom = async () => {
-      const unsanitizedHtml = '<img src="error" onerror="window.document.unsanitizedHtmlFn();">';
-      let isUnsanitizedHtmlRead = false;
-      (editor.getDoc() as any).unsanitizedHtmlFn = () => {
-        isUnsanitizedHtmlRead = true;
+  context('Inserting unparsed content', () => {
+    const assertFnDoesNotReadUnParsedHtmlInDom = async (editor: Editor, fn: (unparsedHtml: string) => void | Promise<void>): Promise<void> => {
+      const fnReadsUnparsedHtmlInDom = async () => {
+        const unparsedHtml = '<img src="error" onerror="window.document.unparsedHtmlFn();">';
+        const isUnParsedHtmlRead = Cell(false);
+        (editor.getDoc() as any).unparsedHtmlFn = () => {
+          isUnParsedHtmlRead.set(true);
+        };
+        await fn(unparsedHtml);
+        // wait for any unparsed html to be read and error to be thrown if it is
+        await Waiter.pWait(1);
+        return isUnParsedHtmlRead.get();
       };
-      await fn(unsanitizedHtml);
-      // wait for any unsanitized html to be read and error to be thrown if it is
-      await Waiter.pWait(1);
-      return isUnsanitizedHtmlRead;
+      assert.isFalse(await fnReadsUnparsedHtmlInDom(), 'Unparsed html read');
     };
-    assert.isFalse(await fnReadsUnsanitizedHtmlInDom(), 'Unsanitized html read');
-  };
 
-  it('TINY-9244: Unsanitized html should not be read when inserting template via command', async () => {
-    const editor = hook.editor();
-    await assertFnDoesNotReadUnsanitizedHtmlInDom(editor, (unsanitizedHtml) => {
-      editor.execCommand('mceInsertTemplate', false, unsanitizedHtml);
+    it('TINY-9244: Unparsed html should not be read when inserting template via command', async () => {
+      const editor = hook.editor();
+      await assertFnDoesNotReadUnParsedHtmlInDom(editor, (unparsedHtml) => {
+        editor.execCommand('mceInsertTemplate', false, unparsedHtml);
+      });
     });
-  });
 
-  it('TINY-9244: Unsanitized html should not be read when inserting template via dialog', async () => {
-    const editor = hook.editor();
-    const unsanitizedHtml = '<img src="error" onerror="window.document.unsanitizedHtmlFn();">';
-    addSettings({
-      templates: [{ title: 'a', description: 'b', content: unsanitizedHtml }],
-    });
-    await assertFnDoesNotReadUnsanitizedHtmlInDom(editor, async () => {
-      await pInsertTemplate(editor);
+    it('TINY-9244: Unparsed html should not be read when inserting template via dialog', async () => {
+      const editor = hook.editor();
+      const unparsedHtml = '<img src="error" onerror="window.document.unparsedHtmlFn();">';
```

Comment:
why don't declare this `unparsedHtml` variable upper in the scope like in the previous `context`

Reply:
That's a good catch, declared in the upper scope.